python/lsst/obs/base/mapper_tests.py

```diff
@@ -0,0 +1,207 @@
+        self.assertEqual(butlerLocation.getStorageName(), "ConfigStorage")
+        processCcd_path = os.path.join(self.mapperData.output, "config", "processCcd.py")
+        self.assertEqual(butlerLocation.getLocations(), [processCcd_path])
+        for k, v in dataId.items():
+            self.assertEqual(butlerLocation.getAdditionalData().get(k), v, msg="Failed for key={}".format(k))
+
+    def test_map_metadata_data(self):
+        dataId = self.dataIds['raw']
+        butlerLocation = self.mapper.map_processCcd_metadata(dataId)
+        self.assertEqual(butlerLocation.getPythonType(), "lsst.daf.base.PropertySet")
+        self.assertEqual(butlerLocation.getCppType(), "PropertySet")
+        self.assertEqual(butlerLocation.getStorageName(), "BoostStorage")
+        self.assertEqual(butlerLocation.getLocations(), [self.mapperData.metadata_output_path])
+        for k, v in dataId.items():
+            self.assertEqual(butlerLocation.getAdditionalData().get(k), v, msg="Failed for key={}".format(k))
+
+    def test_keys(self):
+        self.assertEqual(set(self.mapper.keys()), self.mapperData.keys)
+
+    def test_get_dataset_types(self):
+        someKeys = set(['raw', 'processCcd_config', 'processCcd_metadata'])
+        self.assertTrue(set(self.mapper.getDatasetTypes()).issuperset(someKeys))
+
+    def test_get_keys_raw(self):
+        for level, expect in self.mapperData.raw_levels:
+            result = self.mapper.getKeys("raw", level)
+            self.assertEqual(set(result), expect, msg='Failed for level={}'.format(level))
+
+    def test_get_default_level(self):
+        self.assertEqual(self.mapper.getDefaultLevel(), self.mapperData.default_level)
+
+    def _test_map(self, butlerLocation, dataId):
+        self.assertEqual(butlerLocation.getPythonType(), self.mapperData.map_python_type)
+        self.assertEqual(butlerLocation.getCppType(), self.mapperData.map_cpp_type)
+        self.assertEqual(butlerLocation.getStorageName(), self.mapperData.map_storage_name)
+        locationList = butlerLocation.getLocations()
+        self.assertEqual(len(locationList), 1)
+        fileName = os.path.basename(locationList[0])
+        self.assertEqual(fileName, self.mapperData.raw_filename)
+        for k, v in dataId.items():
+            self.assertEqual(butlerLocation.getAdditionalData().get(k), v, msg="Failed for key={}".format(k))
+
+    def test_map(self):
+        dataId = self.dataIds['raw']
+        self._test_map(self.mapper.map_raw(dataId), dataId)
+        self._test_map(self.mapper.map("raw", dataId), dataId)
+
+    def test_query_metadata(self):
+        """
+        Test expansion of incomplete information of the available data in this
+        obs package's testdata repo.
+        """
+        for query, expect in self.mapperData.queryMetadata:
+            # queryMetadata returns tuples of available items of the 2nd parameter.
+            query = self.mapper.queryMetadata("raw", self.mapperData.query_format, query)
```

Comment:
Clobbering `query`. Prolly doesn't matter, but maybe it gets in the way of debugging?